bitcoin/src/bip32.rs

```diff
@@ -5,12 +5,23 @@
 //! Implementation of BIP32 hierarchical deterministic wallets, as defined
 //! at <https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki>.
 
+use alloc::vec;
+use core::borrow::Borrow;
 use core::ops::Index;
 use core::str::FromStr;
 use core::{fmt, slice};
 
+pub use derivation_path::{
+    DerivationPath, IntoIter as DerivationPathIntoIter, Iter as DerivationPathIter,
+};
+pub use hardened::HardenedChildNumber;
 use hashes::{hash160, hash_newtype, sha512, GeneralHash, HashEngine, Hmac, HmacEngine};
 use internals::{impl_array_newtype, write_err};
+pub use normal::NormalChildNumber;
+pub use normal_derivation_path::{
+    IntoIter as NormalDerivationPathIntoIter, Iter as NormalDerivationPathIter,
+    NormalDerivationPath,
+};
 use secp256k1::{Secp256k1, XOnlyPublicKey};
 
 use crate::crypto::key::{CompressedPublicKey, Keypair, PrivateKey};
```

Comment:
EDIT: Note below I question the non-uniformity of the `NormalDerivationPath` and `DerivationPath` types, same question applies to the module names.

Reply:
>EDIT: Note below I question the non-uniformity of the NormalDerivationPath and DerivationPath types, same question applies to the module names.
You say... the names ? 
>Feel free not to use my custom formating, I grouped similar things instead of using alphabetic order, also one line is longer that the formatter allows.
I dont spend to much time organizing this normally because i know that, usually on many other projects, the fmt will just mess it up or standardize it.